src/main/java/com/google/javascript/clutz/ClutzErrorManager.java

```diff
@@ -0,0 +1,49 @@
+package com.google.javascript.clutz;
+
+import com.google.javascript.jscomp.CheckLevel;
+import com.google.javascript.jscomp.JSError;
+import com.google.javascript.jscomp.MessageFormatter;
+import com.google.javascript.jscomp.PrintStreamErrorManager;
+
+import java.io.PrintStream;
+
+/**
+ * An error manager that delays reporting though {@link #doGenerateReport()}, allows failing fast,
+ * and that adds special, Clutz-specific errors.
+ */
+final class ClutzErrorManager extends PrintStreamErrorManager {
+  private final boolean debug;
+  boolean reportClutzMissingTypes = true;
+
+  /**
+   * @param failEarly if true, directly fail on the first reported error (but not warning!).
+   */
+  ClutzErrorManager(PrintStream stream, MessageFormatter formatter, boolean debug) {
+    super(formatter, stream);
+    this.debug = debug;
+  }
+
+  @Override
+  public void report(CheckLevel level, JSError error) {
+    // Ignore warnings in non-debug mode.
+    if (!debug && level == CheckLevel.WARNING) return;
+
+    if (reportClutzMissingTypes
+        && error.description.contains("Bad type annotation. Unknown type")) {
+      // Prepend an error that hints at missing externs/dependencies.
+      reportClutzMissingTypes = false;
```

Comment:
We still report the error itself, we just *also* report a more specific, more readable message. That message should only appear once.

Added a `fall through` comment to make that more obvious.